src/hrflow_connectors/connectors/leboncoin/connector.py

```diff
@@ -25,36 +26,44 @@
 
 
 def get_job_location(location: t.Dict) -> t.Dict:
-    if location.get("text"):
-        location_text = location["text"]
+    # get location from location dictionary
+    zip_code = None
+    city = None
+    country = None
+    # best case is when location["fields"] is not empty
+    # due to a bug in the API fields is a list instead of a dict
+    if (location["fields"]) is t.Dict:
         zip_code = location["fields"].get("postalcode")
-        print(zip_code)
         city = location["fields"].get("city")
         country = location["fields"].get("country")
-        if zip_code is not None:
-            result_match = re.search("[0-9]{4,5}", location_text)
-            if result_match:
+    # in case zip_code is not in location extract it from the text
+    if zip_code is None:
+        if location["text"]:  # if the text is not an empty string
+            result_match = re.search("[0-9]{4,5}", location["text"])
+            if result_match is not None:
                 zip_code = result_match.group(0)
-    else:
-        zip_code = DEFAULT_ZIP_CODE
+    # else corresponds to the worst case
+    # where zip_code and city could not be found
```

Comment:
So you're not handling the else ? J'ai vu qu'avant tu mettais la valeur par défaut pourquoi pas maintenant